glue-codes/openfast-cpp/src/OpenFAST.H

```diff
@@ -22,53 +22,138 @@
+    int TurbID;
+    //! The FAST Input file name. Typically a .fst file.
+    std::string FASTInputFileName;
+    //! The restart/checkpoint file name. 
+    std::string FASTRestartFileName;
+    //! The position of the base of the turbine in global coordinates
+    std::vector<float> TurbineBasePos;
+    //! The approximate position of the hub of the turbine in global coordinates
+    std::vector<double> TurbineHubPos;
+    //! Number of blades 
+    int numBlades;
+    //! Number of velocity nodes (AeroDyn) per blade
+    int numVelPtsBlade;
+    //! Number of velocity nodes (AeroDyn) on the tower
+    int numVelPtsTwr;
+    //! Total number of velocity nodes (AeroDyn)
+    int numVelPts;
+    //! Desired number of actuator points on each blade
+    int numForcePtsBlade;
+    //! Desired number of actuator points on the tower
+    int numForcePtsTwr;
+    //! Total number of actuator points
+    int numForcePts;
+    //! Inflow Type - 1 (InflowWind) or 2 (Externally specified)
+    int inflowType;
+    //! Drag coefficient of nacelle
+    float nacelle_cd;
+    //! Frontal area of the nacelle
+    float nacelle_area;
+    //! Air density around this turbine
+    float air_density;
 };
 
+//! An id to indicate whether a particular actuator point is on the hub, node or tower    
 enum ActuatorNodeType {
   HUB = 0,
   BLADE = 1,
   TOWER = 2,
   ActuatorNodeType_END
 };
 
+/** An id to indicate the start type of a simulation.
+ * init - Start the simulation from scratch
+ * trueRestart - Restart from a checkpoint file. Code expects checkpoint files for all parts of the simulation including the controller.
+ * restartDriverInitFAST - Start all turbines from scratch and use the velocity data in 'velData.h5' file to run upto desired restart time, then continue the simulation like ''trueRestart'.
+ */
 enum simStartType {
   init = 0,
   trueRestart = 1,
   restartDriverInitFAST = 2,
   simStartType_END    
 };
 
+//! A data structure to hold all velocity and force node information
+struct turbVelForceNodeDataType {
+    //! Blade location at velocity nodes
+    std::vector<double> x_vel;
+    //! Blade velocity at velocity nodes
+    std::vector<double> xdot_vel;
+    //! Sampled velocity at velocity nodes
+    std::vector<double> vel_vel;
+    //! Blade location at force nodes
+    std::vector<double> x_force;
+    //! Blade velocity at force nodes
+    std::vector<double> xdot_force;
+    //! Blade orientation at force nodes
+    std::vector<double> orient_force;
+    //! Sampled velocity at force nodes
+    std::vector<double> vel_force;
+    //! Actuator force at force nodes
+    std::vector<double> force;
+    double x_vel_resid;
+    double xdot_vel_resid;
+    double vel_vel_resid;
+    double x_force_resid;
+    double xdot_force_resid;
+    double orient_force_resid;
+    double vel_force_resid;
+    double force_resid;
+};
+
+//! An enum to keep track of information stored at different time steps
+enum timeStep {
```

Comment:
Descriptive names? `STATE_NM1`, `STATE_NP1` and so on?

Reply:
Agreed. Implemented in 9f34f9eb119d38dc0c78bfe58fdd23d084d02430.